src/dune_rules/scope.ml

```diff
@@ -273,10 +273,24 @@ module DB = struct
           Some (project, stanzas))
       |> Path_source_map_traversals.parallel_map
            ~f:(fun _dir (project, stanzas) ->
-             let+ db =
-               create_db_from_stanzas stanzas ~parent:public_libs ~lib_config
-             in
-             (project, db))
+             match host_context with
+             | None ->
+               let+ db =
+                 create_db_from_stanzas stanzas ~parent:public_libs ~host:None
+                   ~lib_config
+               in
+               (project, db)
+             | Some host_context ->
+               let* scopes, public_libs = create_from_stanzas host_context in
+               let host =
+                 let scope = find_by_project scopes project in
+                 scope.db
+               in
+               let+ db =
+                 create_db_from_stanzas stanzas ~parent:public_libs
```

Comment:
why we are using the public_libs from the host context?

Reply:
looks like a bug. thanks for catching

Reply:
cleaned this up a bit in the most recent push